org.eclipse.xtext.builder/src/org/eclipse/xtext/builder/impl/XtextBuilder.java

```diff
@@ -401,12 +386,17 @@ protected void fullBuild(final IProgressMonitor monitor, boolean isRecoveryBuild
 				: toBeBuiltComputer.updateProject(project, progress.split(2));
 				
 
-		IResourceDeltaVisitor visitor = createDeltaVisitor(toBeBuiltComputer, toBeBuilt, progress);
-		processedAbsentReferencedProjects(visitor);
-		doBuild(toBeBuilt, progress.split(8), 
-			isRecoveryBuild 
-				? BuildType.RECOVERY 
-				: BuildType.FULL);
+		Task task = closedProjectsQueue.exhaust();
+		try {
+			addInfosFromTask(task, toBeBuilt);
```

Comment:
does this make sense in the recovery usecase too?

Reply:
The task contains information about the things that do no longer have the XtextBuilder attached. It will ensure that all the remaining entries are indeed removed from the Xtext index. So yes, I think so.

Reply:
👍